soundsgood.dsp

```diff
@@ -337,36 +337,66 @@ mscomp8i(target) =
   : ms_dec
 ;
 
-
-Eight_band_Compressor_N_chan(N) =
-   crossover
-  : compressors
-  : mixer
+Eight_band_Compressor_N_chan(N) = si.bus (N) <: si.bus (2 * N)
+  : (calcredux(N,(flist)), si.bus (N))
+  : groupGainOut
+  : par(i, N, shelfcascade ((flist)))
   : outputGain
 with {
-  inputGain = par(i, N, _*inGain);
-  crossover =
-    (
-      (crossoverFreqs<:par(i, Nr_crossoverFreqs, _<:si.bus(N)))
-     ,si.bus(N)
-    )
-    <: ro.interleave(N,Nr_bands)
-    :  par(i, N, fi.crossover8LR4)
-    :  ro.interleave(Nr_bands,N);
-  compressors =
-    (strength_array , thresh_array , att_array , rel_array , knee_array , link_array, ro.interleave(N,Nr_bands))
+
+  /* re-order, interleave output:  left gain [Nr_bands .. 1], left-signal , right gain [Nr_bands .. 1], right-signal
+   * Note: shelfcascade takes gains in reverse order */
+  groupGainOut  =
+    (ro.interleave(Nr_bands,N), si.bus(N))
+    : (ro.interleave(N, Nr_bands+1))
+    : par(i, N, ro.cross(Nr_bands),_);
+
+  compressor(N,prePost,strength,thresh,att,rel,knee,link) = co.peak_compression_gain_N_chan (strength,thresh,att,rel,knee,prePost,link,N);
```

Comment:
You are going from db to linear and back.

I would put the following at the bottom of the file, use it here and get rid of the linear2dB before the meter:
```faust:
// +++++++++++++++ peak_compression_gain_mono_db ++++++++++++++++

// TODO: use co.peak_compression_gain_N_chan_db when it arrives in the current faust version

// see: https://github.com/grame-cncm/faustlibraries/pull/140, it's merged in faustlibraries master


peak_compression_gain_mono_db(strength,thresh,att,rel,knee,prePost) =
    abs : ba.bypass1(prePost,si.onePoleSwitching(att,rel)) : ba.linear2db : gain_computer(strength,thresh,knee) : ba.bypass1((prePost !=1),si.onePoleSwitching(rel,att))
with {
    gain_computer(strength,thresh,knee,level) =
        select3((level>(thresh-(knee/2)))+(level>(thresh+(knee/2))),
                0,
                ((level-thresh+(knee/2)) : pow(2)/(2*max(ma.EPSILON,knee))),
                (level-thresh))
        : max(0)*-strength;

Reply:
Thanks, done. 
I see you already have used the same in the zero-layency brickwall limiter. Perhaps after this is merged we can move it into common location, or even a separate file that's included on demand?